core/server/api/canary/posts.js

```diff
@@ -142,50 +145,59 @@ module.exports = {
         permissions: {
             unsafeAttrs: unsafeAttrs
         },
-        query(frame) {
-            return models.Post.edit(frame.data.posts[0], frame.options)
-                .then(async (model) => {
-                    if (!model.get('send_email_when_published')) {
-                        return model;
-                    }
+        async query(frame) {
+            /**Check host limits for members */
+            const hostSettings = config.get('host_settings');
```

Comment:
While the overall limit check of members(the if block) can go in `mega` as a separate method, the use of check itself has to be in the controller as we don't even want to proceed with the `model.edit` request in case the action that user is trying will breach any host set limits. 

So if someone tries publishing the post with `send_email_when_published:true`, we'd ideally want to fail early and not publish the post at all, instead of publishing post and then checking if emails can be sent via mega, failing request if not. An explicit check in controller at top fails the whole request early if the action will break limits. Wdyt? :)

Reply:
Hm :thinking: Think we might want to check the expected behavior with @peterzimon. 
My thinking is, the limit has to do strictly with emails and shout not interrupt the publishing process in any way (same as scheduling).

Reply:
Update: We decided to interrupt publishing/scheduling if the action will breach the host limits and throw an early error with upgrade url